### PR TITLE
Create Project on Accepted Determination

### PR DESCRIPTION
**Pull request by ghickman**
This creates a Project when an accepted Determination is applied to a submission.

I took the opportunity to refactor the `.form_valid()` method to add a fast failure condition to make it clearer that draft submissions are ignored.

Ref #1343

### CODE REVIEW

File: opentech/apply/determinations/views.py [hunk 1]
```diff
@@ -240,27 +241,37 @@ def get_success_url(self):
...
 
-            self.submission.perform_transition(transition, self.request.user, request=self.request, notify=False)
+        self.submission.perform_transition(
+            transition,
+            self.request.user,
+            request=self.request,
+            notify=False,
+        )
+
+        if self.object.outcome == ACCEPTED:
```
Comment (todd-dembrey) [inline opentech/apply/determinations/views.py]: I've realised I missed an acceptance criteria. This should only happen when Accepted and there isnt another stage in the application workflow.

I don't know if we have a method to check that yet, you might want to do some digging. But pseudo code:
```python
 submission.workflow.stages.index(submission.stage) == len(submission.workflow.stages)
```
/musing
Maybe something to add to the workflow class

Comment (ghickman) [reply opentech/apply/determinations/views.py]: Fixed in 8baa461f

**Comment by ghickman** [comment]
@todd-dembrey – added some tests to check determinations create a Project under the correct criteria and added the check you requested.

File: opentech/apply/determinations/tests/test_views.py [hunk 2]
```diff
@@ -126,6 +126,44 @@ def test_can_progress_stage_via_determination(self):
...
+
+        # Cant use refresh from DB with FSM
+        submission_original = self.refresh(submission)
+        submission_next = submission_original.next
+
+        self.assertTrue(hasattr(submission_next, 'project'))
+
+    def test_non_accepted_determination_does_not_create_project(self):
+        submission = ApplicationSubmissionFactory(
+            status='concept_review_discussion',
```
Comment (todd-dembrey) [inline opentech/apply/determinations/tests/test_views.py]: this should be `proposal_review_discussion` the `concept_review_discussion` would fall foul of the `is_final_stage` check, which i cant see being tested anywhere?

Comment (ghickman) [reply opentech/apply/determinations/tests/test_views.py]: What's the difference between these two statuses?

Comment (todd-dembrey) [reply opentech/apply/determinations/tests/test_views.py]: stage 1: `concept_review_discussion` -> `is_final_stage` = False
stage 2: `proposal_review_discussion`-> `is_final_stage` = True

File: opentech/apply/determinations/views.py [hunk 3]
```diff
@@ -260,7 +258,23 @@ def form_valid(self, form):
...
 
-            self.submission.perform_transition(transition, self.request.user, request=self.request, notify=False)
+            self.submission.perform_transition(
+                transition,
+                self.request.user,
+                request=self.request,
+                notify=False,
+            )
+
+            if self.object.outcome == ACCEPTED and self.submission.in_final_stage:
```
Comment (todd-dembrey) [inline opentech/apply/determinations/views.py]: We should check if its in the final stage before peforming the transition.

File: opentech/apply/determinations/tests/test_views.py (hunk 2 above)
Comment (ghickman) [reply opentech/apply/determinations/tests/test_views.py]: Fixed and rebased.

**Pull request merged**